input/pagecontent/algorithms.md

```diff
@@ -1,257 +1,189 @@
-These algorithms are discussed below, and they are also used in the de-identification matrix described in Section 4.8.
+When multiple organizations use the same pseudonymization scheme, they can trade data and perform matching on the pseudonyms. This approach is sometimes called privacy preserving record linkage (PPRL). Some PPRL approaches perform record linkage within a secure data enclave to minimize the risk of unauthorized re-identification. As an alternative, organizations can participate in a private set intersection protocol.
 
-### Redaction
+##### Pseudonyms independent of identifying attributes
 
-Redaction is the process of removing one or more values so that the original information content is no longer observable by human and computer recipients of the data. Redaction is a type of substitution.
+The pseudonym values are created in a way that they are independent of the original personal identifiers. Usually, random values are used to create unique pseudonyms. In order to link the pseudonyms back to it's original personal identifiers, a table (usually called linking table) is used to maintain the relationship between the pseudonyms created independently and the original personal identifiers. The linking table should not be released together with the pseudonymized data, and appropriate technical and organizational measures should be established to protect it from unauthorized access.
 
-#### Characteristics
+##### Pseudonyms derived from identifying attributes using cryptography
 
-Data is fully removed. Risk is minimized.
+Pseudonyms can be cryptographically derived from the values of the attributes that they replace through encryption or hashing.
 
-#### Complete Redaction
+**Encryption** Encrypt the identifiers with a strong encryption algorithm. After encryption, the key can be discarded to prevent decryption. However, if there is a desire to employ the same transformation at a later point in time, the key should be stored in a secure location that is separate from the de-identified dataset. Encryption used for this purpose carries special risks that need to be addressed with specific controls
 
-Some data formats permit complete deletion of both the attribute name and value.
+**Hashing with a keyed hash** A keyed hash is a special kind of hash function that produces different hash values for different keys. The hash key should have sufficient randomness to defeat a brute force attack aimed at recovering the hash key (e.g.,SHA-256 HMAC with a 256-bit randomly generated key). As with encryption, the key should be secret and should be discarded unless there is a desire for repeatability. Hashing used for this purpose carries special risks that need to be addressed with specific controls. Hashing without a key generally does not confer security because an attacker can brute force all possible values to be hashed.
 
-#### Deletion of Value
+#### Transforming of Quasi-Identifiers
```

Comment:
You are introducing a bunch of new data-types, without addressing them on the data-types page. This section seems to be saying the same thing as section 5.8 "De-identification Datatype/Algorithm Matrix" was saying before. I do like the more detail you are providing, but I don't think removing the matrix is helping our reader.